opm/simulators/wells/WellInterfaceGeneric.cpp

```diff
@@ -714,4 +718,74 @@ checkNegativeWellPotentials(std::vector<double>& well_potentials,
     }
 }
 
+void WellInterfaceGeneric::
+updateFiltrationParticleVolume(const double dt, const size_t water_index,
+                               const WellState& well_state, std::vector<double>& filtration_particle_volume) const
+{
+    if (!this->isInjector()) {
+        return;
+    }
+
+    const auto injectorType = this->well_ecl_.injectorType();
+    if (injectorType != InjectorType::WATER) {
+        return;
+    }
+
+    const double conc = this->well_ecl_.getFilterConc();
+    if (conc == 0.) {
+        return;
+    }
+
+    // it is currently used for the filter cake modeling related to formation damage study
+    auto& ws = well_state.well(this->index_of_well_);
+    const auto& connection_rates = ws.perf_data.phase_rates;
+
+    const size_t np = well_state.numPhases();
+    for (int perf = 0; perf < this->number_of_perforations_; ++perf) {
+        // not considering the production water
+        const double water_rates = std::max(0., connection_rates[perf * np + water_index]);
+        filtration_particle_volume[perf] += water_rates * conc * dt;
+    }
+}
+
+void WellInterfaceGeneric::
+updateInjFCMult(const std::vector<double>& filtration_particle_volume) {
+    // the filter injecting concentration, the porosity, the area size
+    // we also need the permeability of the formation, and rw and some other information
+    for (int perf = 0; perf < this->number_of_perforations_; ++perf) {
+        const auto perf_ecl_index = this->perforationData()[perf].ecl_index;
+        const auto& connections = this->well_ecl_.getConnections();
+        const auto& connection = connections[perf_ecl_index];
+        if (this->isInjector() && connection.filterCakeActive()) {
+            const auto& filter_cake = connection.getFilterCake();
+            const double area = connection.getFilterCakeArea();
+            const double poro = filter_cake.poro;
+            const double thickness = filtration_particle_volume[perf] / (area * (1. - poro));
+            const double perm = filter_cake.perm;
+            const double rw = connection.getFilterCakeRadius();
+            const auto cr0 = connection.r0();
+            const auto crw = connection.rw();
+            const auto cskinfactor = connection.skinFactor();
+            const double K = connection.Kh() / connection.connectionLength();
+            const double factor = filter_cake.sf_multiplier;
+            // compute a multiplier for the transmissibility
+            if (filter_cake.geometry == FilterCake::FilterCakeGeometry::LINEAR) {
+                // but we are using this form just for first prototype
```

Comment:
What does this comment mean?

Reply:
It is a leftover from the cleaning up. Will be removed.